marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mastering/default/matching.sjs

```diff
@@ -35,7 +35,7 @@ function filterContentAlreadyProcessed(content, summaryCollection, collectionInf
   const jobIdQuery = cts.fieldWordQuery('datahubCreatedByJob', jobID);
   let auditingNotificationsInSourceQuery = false;
   for (let item of content) {
-    const collections = item.context ? item.context.collections || [] : [];
+    const collections = item.context ? item.context.originalCollections || [] : [];
```

Comment:
This seems like a significant change here, since "originalCollections" is what the document is currently in and "collections" is what the document is supposed to be written to. Has this line always been correct? Or did a change in 5.4 cause this to break?

Reply:
Technically, the matching and mastering steps have set the collections to originalCollections by this point. The change here is to mirror the mastering version that also does locking on document URI. originalCollections makes it more clear that we're interested in the state of collections of the document currently in the database.